crates/da-clients/celestia/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+        // we only need to initiate the http provider and not the ws provider, we don't need async
+
+        let mut headers = HeaderMap::new();
+
+        // checking if Auth is available
+        if let Some(auth_token) = conf.auth_token {
+            let val = HeaderValue::from_str(&format!("Bearer {}", auth_token))?;
+            headers.insert(header::AUTHORIZATION, val);
+        }
+
+        let http_client = HttpClientBuilder::default()
+            .set_headers(headers)
+            .build(conf.http_provider.as_str())
+            .map_err(|e| CelestiaDaError::Client(format!("could not init http client: {e}")))?;
+
+        // Convert the input string to bytes
+        let bytes = conf.nid.as_bytes();
+
+        // Create a new Namespace from these bytes
+        let nid = Namespace::new_v0(bytes)
+            .map_err(|e| CelestiaDaError::Generic(format!("could not init namespace: {e}")))
+            .unwrap();
+
+        Ok(Self { celestia_client: http_client, nid })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use config::{CelestiaDaConfig, DEFAULT_CELESTIA_NODE, DEFAULT_NID};
+
+    #[tokio::test]
+    #[ignore = "Can't run without manual intervention, setup celestia-node and fund address."]
+    async fn test_celestia_publish_state_diff_and_verify_inclusion() {
+        let config = CelestiaDaConfig {
+            http_provider: DEFAULT_CELESTIA_NODE.to_string(),
+            auth_token: None,
+            nid: DEFAULT_NID.to_string(),
+        };
+        // Instantiate CelestiaDaClient
+        let celestia_da_client = CelestiaDaClient::try_from(config).unwrap();
+
+        let s = "Hello World!";
+        let bytes: Vec<u8> = s.bytes().collect();
+        let state_diff = vec![bytes];
+
+        let to: [u8; 32] = [
+            0x01, 0x23, 0x45, 0x67, 0x89, 0xab, 0xcd, 0xef, 0xfe, 0xdc, 0xba, 0x98, 0x76, 0x54, 0x32, 0x10, 0x00, 0x11,
+            0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff,
+        ];
+
+        let height_response = celestia_da_client.publish_state_diff(state_diff, &to).await;
+
+        let height_id = match height_response {
+            Ok(variable) => variable,
+            Err(error) => panic!("Problem reading: {error:?}"),
+        };
+
+        let inclusion_response = celestia_da_client.verify_inclusion(&height_id).await;
+
+        let inclusion = match inclusion_response {
+            Ok(variable) => variable,
+            Err(error) => panic!("Problem reading: {error:?}"),
+        };
+
+        assert_eq!(inclusion, DaVerificationStatus::Verified);
+
+        match inclusion {
+            DaVerificationStatus::Pending => println!("Verification Status is Pending"),
+            DaVerificationStatus::Verified => println!("Verification Status is Verified"),
+            DaVerificationStatus::Rejected(msg) => println!("Verification Status is Rejected: {}", msg),
+        }
```

Comment:
we already have an assert above, is this needed?

Reply:
Valid, can remove.